spec/controllers/submissions_controller_spec.rb

```diff
@@ -129,6 +129,76 @@
       end
     end
 
+    it 'should check for MIME type and extension that match' do
+      expect(@student).to have_accepted_grouping_for(@assignment.id)
+
+      valid_file = fixture_file_upload('docx_file.docx',
+                                       'application/vnd.openxmlformats-officedocument.wordprocessingml.document')
+      content_type = Marcel::MimeType.for Pathname.new(valid_file)
+      file_extension = File.extname(valid_file.original_filename).downcase
+      expected_mime_type = Marcel::MimeType.for extension: file_extension
+
+      expect(content_type).to eq(expected_mime_type)
+
+      post_as @student, :update_files,
+              params: { course_id: course.id, assignment_id: @assignment.id, new_files: [valid_file] }
+      expect(response).to have_http_status :ok
+
+      # update_files action assert assign to various instance variables.
+      # These are crucial for the file_manager view to work properly.
+      expect(assigns(:assignment)).not_to be_nil
+      expect(assigns(:grouping)).not_to be_nil
+      expect(assigns(:path)).not_to be_nil
+      expect(assigns(:revision)).not_to be_nil
+      expect(assigns(:files)).not_to be_nil
+
+      # Check to see if the file was added
+      @grouping.group.access_repo do |repo|
+        revision = repo.get_latest_revision
+        files = revision.files_at_path(@assignment.repository_folder)
+        expect(files['docx_file.docx']).not_to be_nil
+      end
+    end
+
+    it 'should check for MIME type and extension that do not match' do
+      expect(@student).to have_accepted_grouping_for(@assignment.id)
+
+      invalid_file = fixture_file_upload('docx_file.docx', 'application/pdf')
+      allow(Marcel::MimeType)
```

Comment:
Instead of using mocking, create a second test file: take a word document and rename it so that its extension is `.pdf`. Make sure the file name itself communicates that this is what has occurred, so that there's no confusion later. This creates a more authentic test than just mocking alone.